.github/workflows/ci.yml

```diff
@@ -0,0 +1,328 @@
+  # Lint with cabal
+  ################################################################################
+  lint-cabal:
+    name: Lint with cabal
+    runs-on: ubuntu-latest
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v4
+
+    - name: 🛠️ Setup Haskell
+      id: setup-haskell
+      uses:  haskell-actions/setup@v2
+      with:
+        ghc-version: ${{ env.DEFAULT_GHC_VERSION }}
+        cabal-version: ${{ env.DEFAULT_CABAL_VERSION }}
+        cabal-update: false
+
+    - name: 🎗️ Lint with cabal
+      run: ./scripts/lint-cabal.sh
+
+  ################################################################################
+  # Lint with stylish-haskell
+  ################################################################################
+  lint-stylish-haskell:
+    name: Lint with stylish-haskell
+    runs-on: ubuntu-latest
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v4
+
+    - name: 🛠️ Setup stylish-haskell
+      uses: ./.github/actions/setup-stylish-haskell
+      with:
+        stylish-haskell-version: "0.14.6.0"
+        ghc-version: ${{ env.DEFAULT_GHC_VERSION }}
+        cabal-version: ${{ env.DEFAULT_CABAL_VERSION }}
+        # The index-state is fixed to enable caching and ensure that the version
+        # regardless of the current state of Hackage head.
+        # If you want a newer version of stylish-haskell, use a more recent time.
+        hackage-index-state: "2024-04-10T14:36:07Z"
+
+    - name: 🎗️ Lint with stylish-haskell
+      run: ./scripts/format-stylish-haskell.sh && git diff --exit-code
+  
+  ################################################################################
+  # Lint with HLint
+  ################################################################################
+  lint-hlint:
+    name: Lint with HLint
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v4
+
+    - name: 🛠️ Setup HLint
+      uses: haskell-actions/hlint-setup@v2
+      with:
+        version: "3.8"
+
+    - name: 🎗️ Lint with HLint
+      run: ./scripts/lint-hlint.sh
+  
+  ################################################################################
+  # Publish documentation
+  ################################################################################
+  publish-documentation:
+    name: Publish documentation
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' }}
+    needs: [build]
+
+    permissions:
+      pages: write
+      id-token: write
+
+    environment:
+      name: github-pages
+      url: ${{ steps.publish-pages.outputs.page_url }}
+
+    steps:
+      - name: 🛠️ Setup Pages
+        uses: actions/configure-pages@v5
+
+      - name: 📦 Download documentation
+        uses: actions/download-artifact@v4
+        with:
+          name: haddocks-Linux-ghc-${{ env.DEFAULT_GHC_VERSION }}-cabal-${{ env.DEFAULT_CABAL_VERSION }}
```

Comment:
I suppose this would break if we ever decide to build the documentation using a non-default ghc and cabal. Might be useful to add a comment about it?